cpp/cmake_modules/FindClangTools.cmake

```diff
@@ -102,7 +103,27 @@ if (CLANG_FORMAT_VERSION)
                   NO_DEFAULT_PATH
           )
         endif()
+      else()
+        # try searching for "clang-format" and check the version
+        find_program(CLANG_FORMAT_BIN
+          NAMES clang-format
+          PATHS
+                ${ClangTools_PATH}
+                $ENV{CLANG_TOOLS_PATH}
```

Comment:
Do we want to document this in the README?

Reply:
The mechanics of how CMake finds dependencies aren't really documented right now ([looking here](https://github.com/apache/arrow/tree/0ed2b850af7b8606b954256a79d0702c07b275c7/cpp#linting)); I had to read `FindClangTools.cmake` to figure out why my clang-format (which was 6.0, just not named with the prefix) wasn't being accepted.

I think it's out of scope for a general README. Maybe instead there could be a cmake message explaining how clang-format is resolved, displayed if we're in verbose log mode and clang-format isn't found?

Reply:
Or maybe add it to the doccomment?

Reply:
Let's add this to the README since it may come up again. I'll quickly write something